fe/fe-core/src/main/java/org/apache/doris/backup/BackupHandler.java

```diff
@@ -392,11 +410,49 @@ private void restore(Repository repository, Database db, RestoreStmt stmt) throw
         catalog.getEditLog().logRestoreJob(restoreJob);
 
         // must put to dbIdToBackupOrRestoreJob after edit log, otherwise the state of job may be changed.
-        dbIdToBackupOrRestoreJob.put(db.getId(), restoreJob);
+        addBackupOrRestoreJob(db.getId(), restoreJob);
 
         LOG.info("finished to submit restore job: {}", restoreJob);
     }
 
+    private void addBackupOrRestoreJob(long dbId, AbstractJob job) {
+        jobLock.lock();
+        try {
+            Deque<AbstractJob> jobs = dbIdToBackupOrRestoreJobs.computeIfAbsent(dbId, k -> Lists.newLinkedList());
+            if (jobs.size() == Config.max_backup_job_num_per_db) {
+                jobs.removeFirst();
+            }
+            AbstractJob lastJob = jobs.peekLast();
+            // only save the latest job
```

Comment:
Better add more comment to explain why we need to remove the duplicate job at the last of the queue.
I think the reason is because we may add same job with different jobs state when replaying edit log.